pubsub/google/cloud/pubsub_v1/subscriber/client.py

```diff
@@ -49,6 +53,23 @@ class in order to define your own consumer. This is primarily
             arguments.
     """
     def __init__(self, policy_class=thread.Policy, **kwargs):
+        # Use a custom channel.
+        # We need this in order to set appropriate default message size and
+        # keepalive options.
+        target = '{host}:{port}'.format(
+            host=subscriber_client.SubscriberClient.SERVICE_ADDRESS,
+            port=subscriber_client.SubscriberClient.DEFAULT_SERVICE_PORT,
+        )
+        kwargs.setdefault('channel', grpc_helpers.create_channel(
```

Comment:
Would prefer you "do less". i.e. create a `channel` as a single statement and then set the default as another statement. Easier to grok?

Reply:
I do not think it works that way. You can not generate a channel with no target.

Reply:
@lukesneeringer I'm suggesting you make a global named `_TARGET` with this same value rather than re-computing it every time the constructor is called.

Reply:
Down here, I'm suggesting you do:

```python
channel = grpc_helpers.create_channel(
    target=_TARGET,
    scopes=subscriber_client.SubscriberClient._ALL_SCOPES,
    options=(
        ('grpc.max_send_message_length', -1),
        ('grpc.max_receive_message_length', -1),
        ('grpc.keepalive_time_ms', 30000),
    ),
)
kwargs.setdefault('channel', channel)
```

Though it just occurs to me that you probably want to avoid creating `channel` if `'channel' in kwargs` (gRPC may do some things with the channel on creation?)